### ISSUE TITLE
Support dynamic parameters for sliding window aggregation

### ISSUE THREAD

**Issue by Fly-Style** [hazelcast/hazelcast#20383]
Currently, we don't propagate Jet processor's evaluation context correctly to sliding window aggregation operation. 
Generally it works, except dynamic parameters support.

**Comment by viliam-durina** [comment]
I don't know what actually should be fixed here. The arguments of `HOP` or `TUMBLE` arguments are a TABLE, a DESCRIPTOR or an INTERVAL. Neither of these can be passed in as a dynamic parameter. For example `INTERVAL ? SECONDS` doesn't parse in calcite:
```
Incorrect syntax near the keyword 'INTERVAL' at line 1, column 193.
Was expecting one of: ... (a long list)
```
I also tried grouping by parameter so that the parameter would be a part of the `groupKeyFn`, but that's also not allowed. The query was `SELECT window_start, ? FROM ... GROUP BY 1, 2` or `SELECT window_start FROM ... GROUP BY 1, ?`. Both throw:
```
org.apache.calcite.sql.validate.SqlValidatorException: Illegal use of dynamic parameter
```
I tried a couple more queries, but neither failed due to missing evaluation context.
